### PR TITLE
Enforce team slugs to lowercase

### PR DESCRIPTION
**Pull request by stephherbers**
https://github.com/dimagi/open-chat-studio/issues/391

Next step is to auto-generate slugs, but adding this as incremental step

### CODE REVIEW

**Review by SmittieC — COMMENTED** [review]
Nice. I think this warrents a small update to maybe [this test](https://github.com/dimagi/open-chat-studio/blob/0adee2f2837aa24cdf25d1568c74c6cb438e31c0/apps/teams/tests/test_creation.py#L27)?

**Comment by SmittieC** [comment]
I see we're using [get_next_unique_team_slug](https://github.com/dimagi/open-chat-studio/blob/0adee2f2837aa24cdf25d1568c74c6cb438e31c0/apps/teams/helpers.py#L15) to generate one when the user [signs up](https://github.com/dimagi/open-chat-studio/blob/0adee2f2837aa24cdf25d1568c74c6cb438e31c0/apps/teams/helpers.py#L57). Wondering if we shouldn't go with that

**Review by SmittieC — DISMISSED** [review]

File: apps/teams/tests/test_creation.py [hunk 1]
```diff
@@ -24,12 +24,26 @@ def test_create_for_user(self):
         assert [SUPER_ADMIN_GROUP] == [group.name for group in membership.groups.all()]
 
 
-def test_create_team_view(db, client):
+def test_slug_generation_on_team_creation(db, client):
+    """Test to verify tghat the slug is autogenerated when creating a team"""
```
Comment (SmittieC) [inline apps/teams/tests/test_creation.py:28]:
```suggestion
    """Test to verify that the slug is autogenerated when creating a team"""
```

File: apps/teams/tests/test_creation.py [hunk 2]
```diff
@@ -24,12 +24,26 @@ def test_create_for_user(self):
         assert [SUPER_ADMIN_GROUP] == [group.name for group in membership.groups.all()]
 
 
-def test_create_team_view(db, client):
+def test_slug_generation_on_team_creation(db, client):
```
Comment (SmittieC) [inline apps/teams/tests/test_creation.py:27]: I can't seem to find the docs now, but the other day Simon linked me to pytest docs where they recommend that one reserve the `db` fixture for custom fixtures and rather use the `pytest.mark.django_db()` fixture for normal tests

File: apps/teams/tests/test_creation.py [hunk 3]
```diff
@@ -24,12 +25,28 @@ def test_create_for_user(self):
         assert [SUPER_ADMIN_GROUP] == [group.name for group in membership.groups.all()]
 
 
-def test_create_team_view(db, client):
+@pytest.mark.django_db()
+def test_slug_generation_on_team_creation(db, client):
```
Comment (SmittieC) [inline apps/teams/tests/test_creation.py:29]: Since we're now using `@pytest.mark.django_db()`, you can remove the `db` parameter

**Review by SmittieC — APPROVED** [review]

File: apps/teams/forms.py [hunk 4]
```diff
@@ -96,16 +96,21 @@ def save(self, request):
...
-            "slug": _("Team ID"),
         }
         help_texts = {
             "name": _("Your team name."),
-            "slug": _("A unique ID for your team. No spaces are allowed!"),
         }
 
+    def save(self, commit=True):
+        instance = super().save(commit=False)
+        instance.slug = get_next_unique_team_slug(instance.name)
```
Comment (snopoke) [inline apps/teams/forms.py:109]: I think this should only happen if the instance is new. I don't think the slug should ever change once it's been set.

Comment (snopoke) [reply apps/teams/forms.py:109]: https://github.com/dimagi/open-chat-studio/pull/487

**Pull request merged**